pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/performance/AvoidInstantiatingObjectsInLoopsTest.java

```diff
@@ -4,8 +4,98 @@
 
 package net.sourceforge.pmd.lang.java.rule.performance;
 
-import net.sourceforge.pmd.testframework.PmdRuleTst;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
 
-public class AvoidInstantiatingObjectsInLoopsTest extends PmdRuleTst {
-    // no additional unit tests
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sourceforge.pmd.lang.java.ast.ASTAllocationExpression;
+
+public class AvoidInstantiatingObjectsInLoopsTest {
+    AvoidInstantiatingObjectsInLoopsRule rule;
+
+    //CS304 (manually written) Issue link: https://github.com/pmd/pmd/issues/2207
+    @Before
+    public void init() {
+        rule = new AvoidInstantiatingObjectsInLoopsRule();
+    }
+
+    //CS304 (manually written) Issue link: https://github.com/pmd/pmd/issues/2207
+    @Test
+    public void visit() {
+        assertEquals("yes", rule.visit(new ASTAllocationExpression(1), "yes"));
```

Comment:
I'd say, these tests are too fine granular. They do basically a white box testing - they verify that java works as it should, but don't verify, that the rule does what it should...

Please remove your test cases here and create a real test case: https://pmd.github.io/latest/pmd_userdocs_extending_testing.html

The test cases for AvoidInstantiatingObjectsInLoops can be found here:
https://github.com/pmd/pmd/blob/master/pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/performance/xml/AvoidInstantiatingObjectsInLoops.xml

Reply:
Those tests also use a deprecated constructor and are not compatible with the 7.0 branch. Those constructors should never be called manually.